lib/child_process.js

```diff
@@ -183,24 +191,77 @@ function exec(command, options, callback) {
 }
 
 const customPromiseExecFunction = (orig) => {
-  return (...args) => {
-    let resolve;
-    let reject;
-    const promise = new Promise((res, rej) => {
-      resolve = res;
-      reject = rej;
-    });
+  return (file, options = {}) => {
+    let child;
+
+    if (typeof options !== 'object' || options === null) {
+      return PromiseReject(
+        new ERR_INVALID_ARG_TYPE(
+          'options',
+          'Object',
+          options));
+    }
 
-    promise.child = orig(...args, (err, stdout, stderr) => {
-      if (err !== null) {
-        err.stdout = stdout;
-        err.stderr = stderr;
-        reject(err);
-      } else {
-        resolve({ stdout, stderr });
+    const { signal } = options;
+    const hasSignal = signal != null;
+
+    // Lets check if an AbortSignal is passed
+    if (hasSignal) {
+      const isNotValidSignal =
+        typeof signal !== 'object' || !('aborted' in signal);
+
+      if (isNotValidSignal) return PromiseReject(
+        new ERR_INVALID_ARG_TYPE(
+          'options.signal',
+          'AbortSignal',
+          signal));
+
+      // Already aborted
+      if (signal.aborted) return PromiseReject(lazyDOMException('AbortError'));
```

Comment:
I think this is probably not needed since just passing the signal to exec would "automatically" work now that exec supports promises.

Reply:
Sure, done in 0db89e4 👍